Hunter Games: Redux [MDB IGNORE]

Detective-Google opened the pull request:
# About the pull request

### Rewrites the Hunter Games from the ground up.
Inspired by the hunger games, this mode places you in a free for all death match with the other players on the server for the amusement of the Yautja running the game,

###  **There can only be one.**

When the mode is selected, all players (except for predators) are placed into the colony as contestants, and given the directive to be the last survivor, whatever it takes.

### **Latejoins are DISABLED.**

There are (almost) no guns in this mode, melee weapons are scattered throughout the map, with a few scattered crates of stronger bonus gear....

Contestants start with strong skills, a few little starting tools, and a light. Their language is randomized upon spawn out of a selection of human languages-- the curse of babel is cast upon them.

IBs are disabled, all splints are nanosplints, and death is permanent.

During it all, there are Yautja hunters watching, stalking the contestants and amusing themselves with the bloodshed. At the end of it all, this is just to find the most worthwhile prey....

Periodically, supply drops can be voted on by any observers, giving a potentially powerful boon to a contestant.

# Explain why it's good for the game

The hunter games is currently a dreadfully ancient piece of what is effectively rot in our code, it's all written horribly, and it doesn't work, and upon realizing this when trying to fix it up enough to work for events, I took it as a fun challenge to rewrite it.

With the current DDoS paradigm causing pop to occasionally to drop low enough to where players are generally dissatisfied with the HvX gamemode's ability to hold up, I think this will provide a (relatively rare) but interesting breath of fresh air.


# Testing Photographs and Procedure

<details>
<summary>Screenshots & Videos</summary>

testing still in process
![image](https://github.com/user-attachments/assets/c3d19978-8db3-4709-94a3-6aeb3e327f6e)

added sprites ^

</details>


# Changelog

:cl:
add: New (old) Gamemode, the Hunter Games, be the last one standing in a bloody battle royale. Latejoins disabled, Cannot (currently) naturally occur.
refactor: Rewrites the entire hunter games gamemode from the ground up.
maptweak: paths updated for some hunter games spawners, updatepath script included. 
/:cl:

TODO:

- [x] Re-add periodic spectator supply drops
- [x] Work with pred council to iron out what preds can/can't do for the mode
<s>- [ ] Source a sprite for a "yautja" supply drop crate for supply drops to appear in.</s> Didn't have any particularly interested spriters, maybe another time.
- [x] Curate & 'balance' the item drop pools, very random and outdated as of yet.
- [x] Setup/Fix/Do what needs to be done with stats systems-- may require outside expertise.
<s>- [ ] Ensure people don't have to manually set their pref for contestant to high to spawn when readied.</s> to be saved for a follow-up.
- [x] Modifier that disables weapon embedding
<s>- [ ] add spawners to maps that don't have them</s> probably will atomize this so that I dont get stuck in mapping hell with changes to every map in one pr or wait for nanu to have time to review

Detective-Google committed:
experimental huntergames fix

Detective-Google committed:
check fix

Detective-Google committed:
im cooking

Detective-Google committed:
the stove is on fire, the heat is out of control

Detective-Google committed:
more work, roundstart supplies, preset fixes, yautja check

Detective-Google committed:
some cleanup

Detective-Google committed:
little more

Detective-Google committed:
linter

Detective-Google committed:
I'm too lazy to look up how it actually wants it formatted

Detective-Google committed:
more tweaks

Detective-Google committed:
votable

Detective-Google committed:
embed disable

Detective-Google committed:
loot spawner rework

Detective-Google committed:
supply drops, cleanup

Detective-Google committed:
more cleanup

Detective-Google committed:
Merge branch 'master' into huntergames

Detective-Google committed:
remove mode from unsupported maps

Detective-Google committed:
fix bugs from testing

Detective-Google committed:
painkillers

Detective-Google commented:
Drafting for now until I can get around to fixing the yautja checks and doing some more gameplay tests.

Detective-Google committed:
Merge remote-tracking branch 'upstream/master' into huntergames

Detective-Google committed:
Update hunter_games.dm

Detective-Google committed:
screwdriver nerf, loot adjustment, map fix

Detective-Google commented:
seems to be working more or less as intended now.

Detective-Google committed:
fixes the yautja counter, fix github somehow blowing up the almayer during an upstream merge.

Detective-Google committed:
Merge remote-tracking branch 'upstream/master' into huntergames

Detective-Google committed:
update

Detective-Google committed:
debug messages no longer necessary; check fixed.

cm13-github commented:
This pull request has conflicts, please resolve those before we can evaluate the pull request.

Nanu308 reviewed (dismissed):
lgtm

Detective-Google committed:
non votable

Detective-Google committed:
Merge remote-tracking branch 'upstream/master' into huntergames

Detective-Google committed:
actions

Detective-Google commented:
going to be removing the bit to add this to regular rotation because I don't think it's quite ready to run without any intervention & I don't have the time to keep tweaking it just now.

cm13-github commented:
Conflicts have been resolved. A maintainer will review the pull request shortly.

cm13-github commented:
This pull request has conflicts, please resolve those before we can evaluate the pull request.

Detective-Google committed:
Merge remote-tracking branch 'upstream/master' into huntergames

cm13-github commented:
Conflicts have been resolved. A maintainer will review the pull request shortly.

Detective-Google committed:
ok

Detective-Google committed:
Merge remote-tracking branch 'upstream/master' into huntergames

Detective-Google committed:
action icon fix

Detective-Google committed:
Merge remote-tracking branch 'cmss13-devs/master' into huntergames

Detective-Google committed:
icon fix

Detective-Google committed:
Merge remote-tracking branch 'upstream/master'

Detective-Google committed:
Merge remote-tracking branch 'cmss13-devs/master'

Detective-Google committed:
Merge remote-tracking branch 'cmss13-devs/master'

Detective-Google committed:
Merge remote-tracking branch 'origin/master'

Detective-Google committed:
Merge remote-tracking branch 'cmss13-devs/master'

cm13-github commented:
This pull request has conflicts, please resolve those before we can evaluate the pull request.

Detective-Google committed:
Merge remote-tracking branch 'cmss13-devs/master'

Detective-Google committed:
Merge remote-tracking branch 'upstream/master'

Detective-Google committed:
Merge remote-tracking branch 'upstream/master'

Detective-Google committed:
Merge branch 'master' into huntergames

Detective-Google committed:
Merge remote-tracking branch 'upstream/master' into huntergames

Detective-Google commented:
I finally got around to setting up a vm so that I can resolve these when I dont feel like making a process out of it

Detective-Google committed:
github did some weird sh with the maps so im just gonna reset them and run it again

Detective-Google committed:
run the update paths again

Detective-Google committed:
longer

Detective-Google committed:
fix layer visibility thing on pod

Detective-Google committed:
Merge remote-tracking branch 'upstream/master' into huntergames

Detective-Google committed:
dont tell me there's no conflict if there's conflicts

Detective-Google committed:
thank god I can just reset to master and run this again when gh desktop breaks it

cm13-github commented:
This pull request has conflicts, please resolve those before we can evaluate the pull request.

Detective-Google committed:
Merge remote-tracking branch 'upstream/master' into huntergames

cm13-github commented:
Conflicts have been resolved. A maintainer will review the pull request shortly.

Detective-Google committed:
action.dmi resolve

Detective-Google committed:
predators -> yautja hunters

Detective-Google committed:
Fixup maps in TGM format

52be8a8ceeb1: maps/map_files/BigRed/sprinkles/40.viro_open.dmm
52be8a8ceeb1: maps/map_files/DesertDam/Desert_Dam.dmm
52be8a8ceeb1: maps/map_files/LV522_Chances_Claim/LV522_Chances_Claim.dmm
52be8a8ceeb1: maps/map_files/LV624/LV624.dmm
52be8a8ceeb1: maps/map_files/USS_Almayer/USS_Almayer.dmm

Automatically commited by: tools/mapmerge2/fixup.py

Detective-Google committed:
Merge remote-tracking branch 'upstream/master' into huntergames

cm13-github commented:
This pull request has conflicts, please resolve those before we can evaluate the pull request.

Detective-Google committed:
Merge remote-tracking branch 'upstream/master' into huntergames

cm13-github commented:
Conflicts have been resolved. A maintainer will review the pull request shortly.

Detective-Google committed:
do it this way

Detective-Google committed:
Fixup maps in TGM format

b168f6b4c08e: maps/map_files/DesertDam/Desert_Dam.dmm
b168f6b4c08e: maps/map_files/LV522_Chances_Claim/LV522_Chances_Claim.dmm
b168f6b4c08e: maps/map_files/LV624/LV624.dmm
b168f6b4c08e: maps/map_files/USS_Almayer/USS_Almayer.dmm

Automatically commited by: tools\mapmerge2\fixup.py

cm13-github commented:
This pull request has conflicts, please resolve those before we can evaluate the pull request.

Detective-Google commented:
I dont wanna deconflict this anymore I'm tired boss 
<img width="392" height="60" alt="image" src="https://github.com/user-attachments/assets/23693400-3b41-4874-8a22-97dde8ec8417" />
